lib/led/rgb.js

```diff
@@ -475,6 +473,55 @@ RGB.prototype.toggle = function() {
   return this[this.isOn ? "off" : "on"]();
 };
 
+/**
+ * pulse Fade the Led in and out in a loop with specified time
+ * @param  {number} duration Time in ms that a fade in/out will elapse
+ * @return {Led}
+ *
+ * - or -
+ *
+ * @param  {Object} val An Animation() segment config object
+ */
+
+RGB.prototype.pulse = function(duration, callback) {
+  var state = priv.get(this);
+
+  var currentColor = state.prev;
```

Comment:
`pulse()` uses the last color set using `color()`, saved to `state.prev` instead of checking the actual current color of the LEDs.  This is to avoid the problem where subsequent calls to `pulse()` would lead to progressively smaller intensity ranges.  Consider this case:

```
color('red');
pulse(300); // Sets up an animation for intensity 0-100
// 150ms passes
pulse(300); // Current intensity is 50, so new animation created from the interpolated color appears to be intensity 0-50.
// 150ms passes
pulse(300); // Current intensity is 50 but appears to be 25, so new animation appears to be intensity 0-25.
// ...and so on, the animation gets dimmer with each iteration.
```